exporter/splunkhecexporter/logdata_to_splunk.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+func logDataToSplunk(logger *zap.Logger, ld pdata.Logs, config *Config) ([]*splunkEvent, int) {
+	numDroppedLogs := 0
+	splunkEvents := make([]*splunkEvent, 0)
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		if rl.IsNil() {
+			continue
+		}
+
+		ills := rl.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			if ils.IsNil() {
+				continue
+			}
+
+			logs := ils.Logs()
+			for j := 0; j < logs.Len(); j++ {
+				lr := logs.At(j)
+				if lr.IsNil() {
+					continue
+				}
+				ev := mapLogRecordToSplunkEvent(lr, config, logger)
+				if ev == nil {
+					numDroppedLogs++
+				} else {
+					splunkEvents = append(splunkEvents, ev)
+				}
+			}
+		}
+	}
+
+	return splunkEvents, numDroppedLogs
+}
+
+func mapLogRecordToSplunkEvent(lr pdata.LogRecord, config *Config, logger *zap.Logger) *splunkEvent {
+	if lr.Body().IsNil() {
+		return nil
+	}
+	var host string
+	var source string
+	var sourcetype string
+	fields := map[string]string{}
+	lr.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+		if v.Type() != pdata.AttributeValueSTRING {
+			logger.Debug("Failed to convert log record attribute value to Splunk property value, value is not a string", zap.String("key", k))
+			return
+		}
+		if k == conventions.AttributeHostHostname {
+			host = v.StringVal()
+		} else if k == conventions.AttributeServiceName {
+			source = v.StringVal()
+		} else if k == splunk.SourcetypeLabel {
+			sourcetype = v.StringVal()
+		} else {
+			fields[k] = v.StringVal()
```

Comment:
StringVal() won't work if `v` is not a string value. You will probably want to use something like tracetranslator.AttributeValueToString() instead of StringVal(). It is probab;y most important for the last `else` for fields.

Reply:
@tigrannajaryan see a couple lines above, he is only supporting string values at this time and a debug log message will be printed if the attribute is not a string.

Reply:
What Ben says - a bit later in the code I do a mapping for attribute values that is more general.

Reply:
I see, I didn't notice the check.